### PR TITLE
Mhv 52044 new app token for rx

### PR DESCRIPTION
**Pull request by robertbylight**
## Summary

Added a new token to param store that is to be used by the va.gov rx app. Since the mobile team shares the `lib/rx/client.rb` , I needed to create a new client that inherits the existing client but configures the app headers to add the new token. This new client will be used by the  `modules/my_health/app/controllers/my_health/rx_controller.rb`. which is used solely by the va.gov rx app `/my-health/medications/`.

## Related issue(s)

[AAL new token for VA.gov](https://jira.devops.va.gov/browse/MHV-52044)

> Need to update token on the VA.gov side so the AAL entry shows as Remote System: VA.gov instead of VAHB.

## Testing done

manually checked that the token was being retrieved from local development setings.local.yml. However this change will require the [devops](https://github.com/department-of-veterans-affairs/devops/pull/13773) and [vsp-infra-application-manifest](https://dsva.slack.com/archives/CBU0KDSB1/p1702059542262069) prs to get merged first, in order to retrieve the app token for staging, review instance, and prod environments.

## What areas of the site does it impact?

mhv prescriptions controller for rx app used by va.gov site.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -915,6 +916,8 @@ lib/pension_burial @department-of-veterans-affairs/mbs-core-team @department-of-
 lib/periodic_jobs.rb @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 lib/preneeds @department-of-veterans-affairs/mbs-core-team @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 lib/rx @department-of-veterans-affairs/mobile-api-team
+lib/rx/configuration @department-of-veterans-affairs/mobile-api-team @department-of-veterans-affairs/vfs-mhv-medications @department-of-veterans-affairs/backend-review-group
+lib/rx/medications_client.rb @department-of-veterans-affairs/vfs-mhv-medications @department-of-veterans-affairs/backend-review-group
```
Comment (mmoyer-va) [inline .github/CODEOWNERS:920]: Does it make sense to split these out? Should we just one record? e.g. `lib/rx @department-of-veterans-affairs/mobile-api-team @department-of-veterans-affairs/vfs-mhv-medications @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group`

File: lib/common/client/concerns/mhv_session_based_client.rb [hunk 2]
```diff
@@ -84,6 +84,12 @@ def token_headers
         def auth_headers
           config.base_request_headers.merge('appToken' => config.app_token, 'mhvCorrelationId' => session.user_id.to_s)
         end
+
+        protected
+
+        def token_headers_va_gov
+          config.base_request_headers.merge('Token' => session.token)
+        end
```
Comment (mmoyer-va) [inline lib/common/client/concerns/mhv_session_based_client.rb:92]: What is this used for? I don't see it being called from anywhere.

File: .github/CODEOWNERS (hunk 1 above)
Comment (robertbylight) [reply .github/CODEOWNERS:920]: yeah that does make sense, ill go ahead and push that change 👍

**Review by rjohnson2011 — APPROVED** [review]

**Review by aherzberg — APPROVED** [review]
LGTM

**Pull request merged**